apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1163,17 +1152,6 @@ export class RushConfiguration {
     }
   }
 
-  /**
-   * The filename of the build dependency data file.  By default this is
-   * called 'rush-link.json' resides in the Rush common folder.
-   * Its data structure is defined by IRushLinkJson.
-   *
-   * Example: `C:\MyRepo\common\temp\rush-link.json`
-   */
-  public get rushLinkJsonFilename(): string {
```

Comment:
Rather than removing this entirely (which would essentially return `undefined` for legacy code that uses it), perhaps we should mark it as `@deprecated`.  We could change the implementation to throw a more informational error like:
```
throw new Error("The \"rush-link.json\" file was removed in Rush 5.30.0 -- use RushConfigurationProject.localDependencyProjects instead");
```

Then we can remove it for good in the next major release.